code/game/turfs/turf_enter.dm

```diff
@@ -0,0 +1,54 @@
+#define ENTER_PROXIMITY_LOOP_SANITY 100
+// Splitting this into its own proc for profiling purposes.
+/turf/proc/handle_proximity_update(var/atom/movable/mover)
+	var/objects = 0
+	if(!istype(mover) || !(mover.movable_flags & MOVABLE_FLAG_PROXMOVE))
+		return
+	for(var/atom/movable/neighbor in range(1))
+		if(objects > ENTER_PROXIMITY_LOOP_SANITY) 
+			break // Don't let ore piles kill the server as well as the client.
+		objects++
+		mover.HasProximity(neighbor)
+		if(!QDELETED(neighbor) && !QDELETED(mover) && (neighbor.movable_flags & MOVABLE_FLAG_PROXMOVE))
+			neighbor.HasProximity(mover)
+
+#undef ENTER_PROXIMITY_LOOP_SANITY
+/turf/Entered(var/atom/atom, var/atom/old_loc)
```

Comment:
It would be better to type this as a movable instead; none of the logic actually expects this to not be a movable, so the early exit code being mistyped is fine.